### PR TITLE
Fix memory leak related to Torii

### PR DESCRIPTION
**Pull request by luckychess**
Signed-off-by: luckychess <toobwn@gmail.com>

### Description of the Change

This pull request fixes memory leaks in torii_service_test, torii_queries_test and client_test.

### Benefits

Well, leaks are bad ;)

### Possible Drawbacks 

Probably none (I hope)

### Usage Examples or Tests *[optional]*

Build project with ```-fsanitize=address``` and run tests as ```ASAN_OPTIONS=detect_leaks=1 ctest```:
```
90% tests passed, 8 tests failed out of 78

Total Test time (real) =  39.61 sec

The following tests FAILED:
	  3 - ametsuchi_test (Child aborted)
	  4 - wsv_query_command_test (Failed)
	 15 - yac_timer_test (Child aborted)
	 47 - ordering_service_test (Child aborted)
	 49 - ordering_gate_service_test (Child aborted)
	 73 - field_validator_test (Failed)
	 77 - tx_pipeline_integration_test (Child aborted)
	 78 - transfer_asset_inter_domain_test (Child aborted)
```
Without current fix 11 tests will fail.

### P.S.
Please note change in ServerRunner destructor - it doesn't call shutdown for toriiServiceHandler anymore. The reason for that is possible double delete problem because ServerRunner::shutdown also also calls shutdown for toriiServiceHandler.

### CODE REVIEW

**Comment by Warchant** [comment]
Hard to review actual changes, because of applied clang-format. Could you create 2 commits:
1. apply clang-format
2. do the fix

Please.

File: irohad/network/grpc_call.hpp [hunk 1]
```diff
@@ -135,26 +144,36 @@ namespace network {
      * @param requestMethod
      * @param rpcHandler
      */
-    static void enqueueRequest(AsyncService* asyncService,
-                               ::grpc::ServerCompletionQueue* cq,
-                               RequestMethodType requestMethod,
-                               RpcHandlerType rpcHandler) {
+    static CallType *enqueueRequest(AsyncService *asyncService,
```
Comment (muratovv) [inline irohad/network/grpc_call.hpp]: Why do not return `auto` type?

File: irohad/torii/torii_service_handler.cpp [hunk 2]
```diff
@@ -51,35 +51,45 @@ namespace torii {
     while (completionQueue_->Next(&tag, &ok)) {
       // wait until completion queue shuts down
     }
+
+    torii_owner_->responseSent();
```
Comment (muratovv) [inline irohad/torii/torii_service_handler.cpp]: Name is confusing

File: irohad/torii/torii_service_handler.hpp [hunk 3]
```diff
@@ -80,22 +81,30 @@ namespace torii {
...
+              typename RequestType,
               typename ResponseType>
-    void enqueueRequest(
-        network::RequestMethod<AsyncService, RequestType, ResponseType>
-            requester,
-        network::RpcHandler<ToriiServiceHandler, AsyncService, RequestType,
-                            ResponseType>
-            rpcHandler,
-        AsyncService& asyncService) {
+    network::Call<ToriiServiceHandler, AsyncService, RequestType, ResponseType>
```
Comment (muratovv) [inline irohad/torii/torii_service_handler.hpp]: Same issue with auto as before.

Comment (luckychess) [reply irohad/torii/torii_service_handler.hpp]: Can't use auto here because of nullptr (

**Pull request merged**